### PR TITLE
ETK Performance: Defer loading the global styles plugins

### PR DESCRIPTION
**Pull request by arthur791004**
Related to pdDR7T-12w-p2

## Proposed Changes

* Defer loading the global styles plugins
  * jetpack-global-styles-editor-script
  * wpcom-global-styles
  * wpcom-global-styles-editor

## Testing Instructions

1. Checkout this branch.
2. in apps/editing-toolkit, run `yarn dev --sync`.
3. Go to the editor of a sandboxed simple site, and make sure the global styles related feature works as expected

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by taipeicoder — APPROVED** [review]
Still seeing the global styles banners in this branch 👍

**Comment by BogdanUngureanu** [comment]
I think we need to use `wp_script_add_data( 'foo', 'strategy', 'defer' );` for BC, right? :)

**Review by rcrdortiz — DISMISSED** [review]
Thanks for taking care of this! The code looks good and Global Styles are working as expected on the Post editor and Site editor.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php [hunk 1]
```diff
@@ -213,10 +212,9 @@ function wpcom_global_styles_enqueue_assets() {
 		plugins_url( 'dist/wpcom-global-styles-view.min.js', __FILE__ ),
 		$dependencies,
 		$version,
-		array(
-			'strategy' => 'defer',
-		)
+		true
```
Comment (rcrdortiz) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: Shouldn't these be false? Well, it's a bit ambiguous, I think it makes sense to load them on the footer. But then the code we had before would be equivalent to your previous commit according to the docs 🤔

Comment (arthur791004) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: Oh, it's a good question. I didn't notice we put it in the footer...and this change might not improve the performance 😂

Comment (arthur791004) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: @alshakero I think we don't need to apply changes to the scripts in the footer, right?

Comment (alshakero) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: Yes we do, footer scripts will still block the main thread during the first render, while deferred scripts won't.

Comment (arthur791004) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: The editor is rendered by the initializeEditor function, and [it's called when the DOMContentLoaded event fired](https://github.com/WordPress/wordpress-develop/blob/ffcf5fb38aa0de123f71ef89d14de77eaa5904c0/src/wp-admin/site-editor.php#L122).

> The DOMContentLoaded event fires when the HTML document has been completely parsed, and all deferred scripts ([<script defer src="…">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer) and [<script type="module">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#module)) have downloaded and executed. It doesn't wait for other things like images, subframes, and async scripts to finish loading.

Referring to [here](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) and [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer), the `DOMContentLoaded` event still waits for the deferred scripts. Hence, deferring ETK scripts won't make the editor be initialized earlier CMIIW.

Comment (alshakero) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php:215]: > The editor is rendered by the initializeEditor function, and [it's called when the DOMContentLoaded event fired](https://github.com/WordPress/wordpress-develop/blob/ffcf5fb38aa0de123f71ef89d14de77eaa5904c0/src/wp-admin/site-editor.php#L122).

This changes everything. I'll reach out to PerfOps.

**Comment by arthur791004** [comment]
Closing it as it's not planned. See pdDR7T-12w-p2#comment-1329.

**Pull request closed without merging**